From d2bdfc5939930af262f29660f1e74e80a46bd52e Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Mon, 14 Apr 2014 03:59:54 -0400
Subject: [PATCH] pyanaconda/__init__.py: check ksdata.liveimg before import
 installclass

The wrlinux.py checks flags.livecdInstall in the __init__(), we need set a
proper flags.livecdInstall before import it.

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 pyanaconda/__init__.py |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/pyanaconda/__init__.py b/pyanaconda/__init__.py
index b38bd78..d092318 100644
--- a/pyanaconda/__init__.py
+++ b/pyanaconda/__init__.py
@@ -122,6 +122,9 @@ class Anaconda(object):
     @property
     def instClass(self):
         if not self._instClass:
+            if self.ksdata and self.ksdata.liveimg.liveimg:
+                from flags import flags
+                flags.livecdInstall = True
             from installclass import DefaultInstall
             self._instClass = DefaultInstall()
 
-- 
1.7.10.4

